core/chain-configs/src/genesis_config.rs

```diff
@@ -586,6 +594,21 @@ impl Genesis {
             }
         }
     }
+
+    /// Forces loading genesis records into memory.
+    ///
+    /// This is meant for **tests only**.  In production code you should be
+    /// using [`Self::for_each_record`] instead to iterate over records.
+    ///
+    /// If the records are already loaded, simply returns mutable reference to
+    /// them.  Otherwise, reads them from `records_file`, stores them in memory
+    /// and then returns mutable reference to them.
+    pub fn force_read_records(&mut self) -> &mut GenesisRecords {
+        if self.records.as_ref().is_empty() {
+            self.records = GenesisRecords::from_file(&self.records_file);
+        }
+        &mut self.records
```

Comment:
I wonder if just 

```
    pub fn records_mut(&mut self) -> &mut GenesisRecords {
        assert!(!self.records.as_ref().is_empty());
        &mut self.records
    }
```

would work here -- just erroring out if the records are not in-memory. Using this function in tests where records are on the side feels like a bug.

Didn't check whether this'd work.

Reply:
It most likely would work but I’m not sure if we’re winning anything with that.

Reply:
The current code makes it seem that loading records from file and then modifying them is a reasonable, useful thing to do. "You can only modify in-memory genesis" is much more constrained, which would help the reader understand the intended use-case of this functionality better. Minor point